### PR TITLE
Hub 01 intercom dialogue revamp

### PR DESCRIPTION
**Pull request by Ilysen**
#### Summary
Content "The dialogue of Hub 01's intercom has been updated."

#### Purpose of change

I felt like the intercom dialogue could use a touch-up. From an in-character perspective, it felt wrong that you have to genuinely plead with them for them to open up to working together, and as a general rule of thumb, the writing had a lot of grammar issues. They're shady bastards, but some of their dialogue felt remarkably passive-aggressive for a group of researchers that I feel would be able to see the practicality in an agreement of some kind.

#### Describe the solution

* Rewrote the majority of dialogue for the intercom. They maintain their forcefulness about certain things, but I've tried to make it clear that they're still human and they aren't flat-out evil, even if they have a shady agenda and strange practices. This isn't to say they're good; I've just endeavored to lay some more foundation for the player to build an image of the group in their head.
* Broke up the huge `NPC_ROBOFAC_INTERCOM.json` file into multiple subfiles for different topics, such as the intro sequence or dialogue that's only for fluff. The main file remains, though, as it houses the definitions for the NPC itself and all the dialogue that doesn't necessitate its own file.
* The intro quest (`Retrieve Field Data`) is functionally the same for the player, but is no longer classified as a fetch quest in the code, instead being assigned and completed via talk_topic effect calls.
* Added a new dialogue tree to discuss their theories on portal storms, and to serve as an acknowledgement by NPCs in the world that they're actually happening. The option becomes available after you take the `Steal a dead man's mind` quest, whose description mentions portal storms, but another path to the tree will also become available as soon as you witness a portal storm firsthand.
* All other quests are untouched in both writing and code.
* Shop inventory is untouched, as are the Free Merchants that show up outside after finishing `Steal a dead man's mind`. In the future, I'd like to tie that to its own quest, but I figured it was out of scope..
* Many portions of dialogue are untouched from their current iterations. This just changes the standard dialogue you get from interacting with the intercom during the intro and through some of the options in the main menu.

This is CR because I'd like to get some input from @John-Candlebury, and to make sure that the way I handled the refactor of the intro quest isn't a cardinal sin against code (setting the goal to `MGOAL_NULL`.) Finally, I really don't know if there are proper naming conventions for vars added by `u_add_var` and related calls, so I'm trying to touch base on that.

#### Describe alternatives you've considered

I wanted to do a few other things in this PR, and revamp the dialogue for the prototyping system as well as the other quests, but I decided it was out of scope. Right now, I'm using this as a proof-of-concept and to lay the foundation for further work to be done on the intercom.

I also wanted to create dialogue to "pick up" on choices that the player backed out of at an earlier point. You can see this in the hard drive trade, where the dialogue is different if you had already entered the tree versus if you're doing it for the first time. This is another thing I wanted to do, but due to the overhead of how much sheer dialogue there is to account for, I stuck with one or two instances for now.

#### Testing

I jumped to Hub 01 using the debug menu and did a bunch of talking and dialogue chains. Debug was used to spawn in quest items when necessary, but I also tested going to the locations manually, grabbing the item, and teleporting back.

#### Additional context

image
cataclysm-tiles_q5uLnKDCSA
cataclysm-tiles_MfQNTmIVzg

### CODE REVIEW

File: data/json/npcs/robofac/intercom/robofac_intercom_prototypes.json [hunk 1]
```diff
@@ -257,7 +257,7 @@
...
-    "dynamic_line": "Yes, we've made great progress!  I'll spare you the details but we've developed much more resilient materials for our armors while sacrificing no maneuverability.  As a thank you for bringing this to us the first working model is yours merc.  That said these do take a considerable amount of high-end materials.  If you want the matching pieces we'll need 6 of our coins to barely cover the materials cost.  Also, I've been told to inform you we would be happy to collaborate like this again in the future.",
+    "dynamic_line": "Yes, we've made great progress!  The data on the template was invaluable.  I'll spare you the details, it allowed us to produce much more resilient materials for the attachments without sacrificing the agility.  Testing is still ongoing, but-\"  The voice quickly confirms something off-mic.  \"We are willing to provide you with the first working model, but these prints take a considerable amount of high-end materials.  If you want the matching pieces, we would like a flat fee of 6 HGC to cover the materials cost.\"\n\n\"I've also been told to inform you that we would be happy to collaborate like this again in the future.",
```
Comment (bombasticSlacks) [inline data/json/npcs/robofac/intercom/robofac_intercom_prototypes.json:260]: I don't think "the agility" sounds less awkward than maneuverability here. It also IMO sounds less like someone with modern combat training.

File: data/json/npcs/robofac/intercom/robofac_intercom_prototypes.json [hunk 2]
```diff
@@ -302,6 +302,11 @@
       }
     ]
   },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_ARMOR_BREAKTHROUGH_END",
+    "type": "talk_topic",
+    "dynamic_line": "Thank you.  The mantle and skirt will clip to your MDS, and the other pieces can be worn standalone.  If you happen upon anything else like this in the future, we would be happy to collaborate again."
```
Comment (bombasticSlacks) [inline data/json/npcs/robofac/intercom/robofac_intercom_prototypes.json:308]: This is a great addition, and it looks like it goes to here in all the right places 👍

File: data/json/npcs/robofac/intercom/robofac_intercom_hdd_trade.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      "TALK_ROBOFAC_INTERCOM_FREE_MERCHANT_DELIVERY_3"
+    ],
+    "type": "talk_topic",
+    "dynamic_line": {
+      "u_has_var": "u_started_robofac_hdd_trade",
+      "type": "dialogue",
+      "context": "intercom",
+      "value": "yes",
+      "yes": "&The intercom's tray wordlessly slides open.",
+      "no": "&The intercom barks as a voice clears its throat, then speaks slowly and carefully.  \"Understood.  There is a tray embedded beneath the intercom.  Please place the drive into it.\"  On cue, a small portion of the wall beneath the intercom slides out along rails, revealing a large tray several inches deep.  \"You are welcome to leave afterwards.\""
```
Comment (John-Candlebury) [inline data/json/npcs/robofac/intercom/robofac_intercom_hdd_trade.json:16]: Hmm I wouldn't describe the intercom using biological terms. I know its a pretty small detail, but this is potentially the first interaction a new player get with the faction, so i think its important here. I'd suggest something like:

"An embed speaker erupts into static before resolving to a slow speaking voice"
Although there are other lines like that already hmm.

**Comment by Leland** [comment]
> Keep in mind renaming and moving all these files completely breaks git blame

I agree with this – file name and placement is a lower concern than maintaining a readable history.

It also makes reviewing the changed lines in this PR difficult. If you still want to split files out (or rename/move them), I'd strongly urge doing that in a separate, follow-up PR. 

Thanks for working on dialogue, though! Love to see it.

**Comment by Ilysen** [comment]
alright; should I consolidate everything back into the main file here, then?

**Comment by Leland** [comment]
What I would do in your shoes is drop all the commits in this PR so far and force push a new, single commit with all the changes applied on the original files.

<details>
<summary>Here's an example of how to reset your branch (apologies if you already know this): </summary>
1. Make sure to stash or commit any uncommitted work, this will wipe the slate clean: 

    git checkout robofac-dialogue-audit && git reset --hard master
2. Then, just replay your changes onto the files (copy/pasting – or something more creative), create a new commit, and force push that up to your branch!

ℹ️ if you ever want to return to the state of your branch in this PR today, do `git reset --hard 4427978`
</details>

Then, once this PR has gone through review and been merged, open a second PR (if you feel so inclined!) with your proposed file system changes, where reviewers can discuss that.

**Comment by Ilysen** [comment]
Mmkay, can do. Might be a bit tho.

**Comment by Ilysen** [comment]
Not sure if there's anything else y'all see that needs to be touched up?

**Comment by John-Candlebury** [comment]
All seems right to me!

**Pull request merged**